imjoy/worker3.py

```diff
@@ -0,0 +1,107 @@
+"""Provide worker functions for Python 3."""
+import inspect
+import sys
+import traceback
+
+from .util import Registry, make_coro
+from .worker import JOB_HANDLERS
+
+# pylint: disable=unused-argument
+
+JOB_HANDLERS_PY3 = Registry()
+JOB_HANDLERS_PY3.update({name: make_coro(func) for name, func in JOB_HANDLERS.items()})
+
+
+async def task_worker(conn, async_q, logger, abort=None):
+    """Implement a task worker."""
+    while True:
+        if abort is not None and abort.is_set():
+            break
+        job = await async_q.get()
+        if job is None:
+            continue
+        handler = JOB_HANDLERS_PY3.get(job["type"])
+        if handler is None:
+            continue
+        try:
+            await handler(conn, job, logger)
+        except Exception:
+            print("error occured in the loop.", traceback.format_exc())
+        finally:
+            sys.stdout.flush()
+            async_q.task_done()
+
+
+@JOB_HANDLERS_PY3.register("method")
+async def handle_method_py3(conn, job, logger):
+    """Handle method."""
+    if job["name"] in conn.interface:
+        if "promise" in job:
+            try:
+                resolve, reject = conn.unwrap(job["promise"], False)
+                method = conn.interface[job["name"]]
+                args = conn.unwrap(job["args"], True)
+                # args.append({'id': conn.id})
+                result = method(*args)
+                if result is not None and inspect.isawaitable(result):
+                    result = await result
+                resolve(result)
+            except Exception as exc:  # pylint: disable=broad-except
+                logger.error(
+                    "error in method %s: %s", job["name"], traceback.format_exc()
+                )
+                reject(exc)
```

Comment:
Same here, to see the traceback in ImJoy, change to `reject(traceback.format_exc())`